File: medium/halloween-number/ROT13Cipher.py
def encrypt(plainText):
    result = ""

    for i in range(len(plainText)):
        char = plainText[i]

        if (char.isupper()):
            result += chr((ord(char) + 13 - 65) % 26 + 65)

        else:
            result += chr((ord(char) + 13 - 97) % 26 + 97)

    return result

def decrypt(cipherText):
    result = ""

    for i in range(len(cipherText)):
        char = cipherText[i]

        if (char.isupper()):
            result += chr((ord(char) - 13 - 65 + 26) % 26 + 65)

        else:
            result += chr((ord(char) - 13 - 97 + 26) % 26 + 97)

    return result

question = input('Do you want to encrypt or decrypt text? (encrypt/decrypt) ')

if question == 'encrypt':
    plainText = input('Enter the plaintext that you would like to encrypt: ')

    print("Encrypting '" + plainText + "' ...")

    encryptedText = encrypt(plainText)
    print("Encrypted Text: " + encryptedText)
elif question == 'decrypt':
    cipherText = input('Enter the ciphertext that you would like to decrypt: ')
    
    print("Decrypting '" + cipherText + "' ...")
        
    decryptedText = decrypt(cipherText)
    print("Decrypted Text: " + decryptedText)

File: medium/halloween-number/ROT13Cipher.cs
using System;

public class ROT13Cipher {
    public static StringBuilder encrypt(String plaintext)
    {
        StringBuilder result= new StringBuilder();

        for (int i=0; i<plaintext.Length; i++) {
            if (char.IsUpper(plaintext[i])) {
                char ch = (char)(((int)plaintext[i] + 13 - 65) % 26 + 65);
                result.Append(ch);
            } else {
                char ch = (char)(((int)plaintext[i] + 13 - 97) % 26 + 97);
                result.Append(ch);
            }
        }
        return result;
    }
    public static StringBuilder decrypt(String ciphertext)
    {
        StringBuilder result= new StringBuilder();

        for (int i=0; i<ciphertext.Length; i++) {
            if (char.IsUpper(ciphertext[i])) {
                char ch = (char)(((int)ciphertext[i] - 13 + 26 - 65) % 26 + 65);
                result.Append(ch);
            } else {
                char ch = (char)(((int)ciphertext[i] - 13 + 26 - 97) % 26 + 97);
                result.Append(ch);
            }
        }
        return result;
    }

    public static void Main(String[] args) {
        Console.WriteLine("Do you want to encrypt or decrypt text? (encrypt/decrypt)");
        string question = Console.ReadLine();

        switch(question) {
            case "encrypt";
                Console.WriteLine("Enter the plaintext that you would like to encrypt:");
                string plaintext = Console.ReadLine();

                Console.WriteLine("Encrypting '" + plaintext + "' ...");
                string encryptedText = encrypt(plaintext);

                Console.WriteLine("Encrypted Text: " + encryptedText);
                break;
            case "decrypt";
                Console.WriteLine("Enter the ciphertext that you would like to decrypt:");
                string ciphertext = Console.ReadLine();

                Console.WriteLine("Encrypting '" + ciphertext + "' ...");
                string decryptedText = decrypt(ciphertext);

                Console.WriteLine("Decrypted Text: " + decryptedText);
                break;
        }
    }
}
